pkg/optimization/regex.go

```diff
@@ -0,0 +1,68 @@
+package optimization
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	REGEXP_LINE_MAINSTAT = regexp.MustCompile(`(?m)^[a-z]+\s+add\s+stats\s+hp=(4780|3571)\b[^;]*;`)
+	REGEXP_LINE_SUBSTAT  = regexp.MustCompile(`(?m)^[a-z]+\s+add\s+stats\b[^;]*;.*\n`)
+	REGEXP_LINE_CHARNAME = regexp.MustCompile(`(?m)^([a-z]+)\s+char\b[^;]*;`)
+	REGEXP_LINE_OPTIONS  = regexp.MustCompile(`([a-z_]+)=([0-9.]+)`)
+)
+
+type OptimRegex struct {
+	Mainstats      *regexp.Regexp
+	GetCharNames   *regexp.Regexp
+	Substats       *regexp.Regexp
+	Options        *regexp.Regexp
+	InsertLocation *regexp.Regexp
+}
+
+var InvalidStats = errors.New("Error: Could not identify valid main artifact stat rows for all characters based on flower HP values.\n5* flowers must have 4780 HP, and 4* flowers must have 3571 HP.")
```

Comment:
error messages should really just be a short sentence, all lower case (no new line). for example i would have just wrote for this one `var ErrInvalidStats = errors.New("unidentifiable main stats")`

also i believe the convention in go code is to name these variables starting with `Err`